ewatercycle/models/lisflood.py

```diff
@@ -0,0 +1,398 @@
+            args.append(docker_image)
+
+        else:
+            raise ValueError(
+                f"Unknown container technology in CFG: {CFG['container_engine']}"
+            )
+
+        lisvap_filename = Path(lisvap_file).name
+        args += ['python3', '/opt/Lisvap/src/lisvap1.py', f"/output/{lisvap_filename}"]
+        container = subprocess.Popen(args, preexec_fn=os.setsid, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        exit_code = container.wait()
+        stdout, stderr = container.communicate()
+        return exit_code, stdout, stderr
+
+    def get_value_as_xarray(self, name: str) -> xr.DataArray:
+        """Return the value as xarray object."""
+        # Get time information
+        time_units = self.bmi.get_time_units()
+        grid = self.bmi.get_var_grid(name)
+        shape = self.bmi.get_grid_shape(grid)
+
+        # Extract the data and store it in an xarray DataArray
+        da = xr.DataArray(
+            data=np.reshape(self.bmi.get_value(name), shape),
+            coords={
+                "longitude": self.bmi.get_grid_y(grid),
+                "latitude": self.bmi.get_grid_x(grid),
+                "time": num2date(self.bmi.get_current_time(), time_units)
+            },
+            dims=["latitude", "longitude"],
+            name=name,
+            attrs={"units": self.bmi.get_var_units(name)},
+        )
+
+        return da
+
+    @property
+    def parameters(self) -> Iterable[Tuple[str, Any]]:
+        """List the parameters for this model."""
+        if self.forcing_dir and self.work_dir:
+            return {
+                'forcing_dir': self.forcing_dir,
+                'work_dir': self.work_dir,
+            }.items()
+        return []
+
+
+def reindex_forcings(mask_map: PathLike, forcing: ForcingData, output_dir: PathLike = None) -> PathLike:
+    """Reindex forcing files to match mask map grid
+
+    Args:
+        mask_map: Path to NetCDF file used a boolean map that defines model boundaries.
+        forcing: Forcing data from ESMValTool
+        output_dir: Directory where to write the re-indexed files, given by user or created for user
+
+    Returns:
+        Output dir with re-indexed files.
+    """
+    output_dir = _generate_workdir(output_dir)
+    mask = xr.open_dataarray(mask_map).load()
+    data_files = list(forcing.recipe_output.values())[0].data_files
+    for data_file in data_files:
+        dataset = data_file.load_xarray()
+        out_fn = output_dir / data_file.filename.name
+        var_name = list(dataset.data_vars.keys())[0]
+        encoding = {var_name: {"zlib": True, "complevel": 4, "chunksizes": (1,) + dataset[var_name].shape[1:]}}
+        dataset.reindex(
+                    {"lat": mask["lat"], "lon": mask["lon"]},
+                    method="nearest",
+                    tolerance=1e-2,
+                ).to_netcdf(out_fn, encoding=encoding)
+    return output_dir
+
+
+def _generate_workdir(work_dir: PathLike = None) -> PathLike:
```

Comment:
Why is this a function rather than a method?

Reply:
The `_generate_workdir` is also used in `reindex_forcings` that is not part of the model class.

Reply:
See my new comment on this usage from withing `reindex_forcings`. It seems it's not doing anything. I still don't understand.

Reply:
right, the function `_check_work_dir` is removed now.